src/client/opamCommands.ml

```diff
@@ -3663,7 +3667,7 @@ let clean cli =
     in
     let switches =
       if all_switches then OpamGlobalState.switches gt
-      else if switch then match OpamStateConfig.get_switch_opt () with
+      else if switch || untracked then match OpamStateConfig.get_switch_opt () with
```

Comment:
One could call `opam clean --untracked --all-switches`
```suggestion
      else if switch then match OpamStateConfig.get_switch_opt () with
```

Reply:
It is already handled by the line above
```
       if all_switches then OpamGlobalState.switches gt 
```

Reply:
Ah my bad